lib/query-builder.js

```diff
@@ -3139,6 +3139,80 @@ function TermDef(textQuery) {
   }
 }
 util.inherits(TermDef, QueryDef);
+
+function before() {
+   var args = mlutil.asArray.apply(null, arguments);
+   if(args.length === 0 || args[0] === null)
+     throw new Error('missing Timestamp');
+   if(args.length>1)
+     throw new Error('Only one timestamp is allowed');
+   var arg = new String(args[0]);
```

Comment:
Would it make sense to pass arg[0] without wrapping in new String() to the TimestampDef constructor and move the arg testing and conversion into the TimestampDef constructor so we maintain that code in one spot and so the TimestampDef object is fully responsible for its data?

Should the TimestampDef constructor allow for either a string primitive or a String object without conversion (in particular, without constructing a new String() object)?

A string primitive and a String object are different types in JavaScript.  For an example, the following code snippet:

```
const primitive = "value";
const object    = new String(primitive);
[typeof primitive, primitive instanceof String, typeof object, object instanceof String];
```

yields `["string", false, "object", true]`

If so, the TimestampDef constructor would skip conversion for if arg instanceof String || typeof arg == 'string'